File: _frameworks/vienna/lib/runtime/classes/string.js
// 
//  string.js
//  vienna
//  
//  Created by Adam Beynon on 2009-10-02.
//  Copyright 2009 Adam Beynon. All rights reserved.
// 

Object.extend(String, Object.VNCoreMethods);

String.extend({
  
  capitalize: function() {
    return this.charAt(0).toUpperCase() + this.substr(1);
  },
  
  w: function() {
    return this.split(' ');
  },
  
  toArray: function() {
    return this.split('');
  },
  
  format: function() {
    var i = 0, args = arguments;
    return this.replace(/%@/g, function(s) {
      if (i < args.length) {
        var ret = args[i];
        i++;
        return ret.toString();
      }
      else {
        return '(null)';
      }
    });
  }
});

File: core/string.rb
# 
#  string.rb
#  vienna
#  
#  Created by Adam Beynon on 2010-06-30.
#  Copyright 2010 Adam Beynon. All rights reserved.
# 

class String
  
  def initialize(string="")
    `#{self}.__str__ = #{string};`
  end
  
  def == other
    `return #{self}.__str__ === #{other}.__str__;`
  end
  
  def upcase!
    `return #{self}.__str__ = #{self}.__str__.toUpperCase();`
  end
  
end

`console.log("===== String test");`
string_test = "hey there"
`console.log(#{string_test});`
